### PR TITLE
Improve Bundler SDK release pipeline

### LINKED ISSUE #780: [AA] Automatic Version Increment, Release on Develop commit (*with* changes)
## Issue Type

```
[ ] bug report
[ X] feature request
```

## Current Behavior
Currently we create a new Github release and need to change the version number in package.json for the bundler-sdk manually. 

## Expected Behavior
Auto-increment version number (revision), minor & major have to be updated manually? [TBD] -> maybe there is a good extension/tool for that. 

Only do a new release when there is a changeset for this code and eventually also add this for merges that there will be a release or no release when there is a/no changeset. 
image

Reference to these comments: 
https://github.com/bobanetwork/boba/pull/749#discussion_r1184842810

### PR DESCRIPTION
**Pull request by wsdt**
:clipboard: resolves #780

## Overview

Improving bundler sdk release pipeline.

## Changes

Describe your changes and implementation choices. More details make PRs easier to review.

- Fixed @boba -> @bobanetwork scope as boba-scope is taken on npmjs
- Changed action triggers & added path filter (only changeset)
- Auto increment minor version and adapt local dependencies to not break when major, minor or patch is released. 
- Add PR commenter when there is a changeset in the `bundler_sdk` directory. 

## Testing

Test release via Github actions (package removed again)

### CODE REVIEW

File: package-lock.json
Comment (souradeep-das) [inline package-lock.json:1]: leftover?

File: yarn.lock
Comment (souradeep-das) [inline yarn.lock:1]: looks like this changed a lot for a rename, can we recheck? 😄

**Review by souradeep-das — APPROVED** [review]
nice!

File: yarn.lock
Comment (wsdt) [reply yarn.lock:1]: pipeline ran through on my end and compiled :-)

File: package-lock.json
Comment (wsdt) [reply package-lock.json:1]: done!

**Review by InoMurko — APPROVED** [review]

**Pull request merged**